crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -1762,21 +1762,29 @@ pub(super) fn report_possibly_unbound_attribute(
     ));
 }
 
-pub(super) fn add_inferred_python_version_hint(db: &dyn Db, mut diagnostic: LintDiagnosticGuard) {
+/// Add a subdiagnostic to `diagnostic` that explains why a certain Python version was inferred.
+///
+/// ty can infer the Python version from various sources, such as command-line arguments,
+/// configuration files, or defaults.
+pub fn add_inferred_python_version_hint_to_diagnostic(
+    db: &dyn Db,
+    diagnostic: &mut Diagnostic,
+    action: impl std::fmt::Display,
```

Comment:
To avoid monomorphization (this isn't perf sensitive and it involves a fair amount of code)

```suggestion
    action: &dyn std::fmt::Display,
```

Reply:
I'll just change this to `&str` for now; we can switch it to a more general type if we ever actually have a need for it!